avm/ptn/ai-platform/baseline/main.bicep

```diff
@@ -61,6 +96,177 @@ resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableT
+        }
+      ]
+    }
+  }
+]
+
+module defaultNetworkSecurityGroup 'br/public:avm/res/network/network-security-group:0.3.1' = if (createDefaultNsg) {
+  name: '${uniqueString(deployment().name, location)}-nsg'
+  params: {
+    name: 'nsg-${name}'
+    location: location
+    securityRules: [
+      {
+        name: 'DenySshRdpOutbound'
+        properties: {
+          priority: 200
+          access: 'Deny'
+          protocol: '*'
+          direction: 'Outbound'
+          sourceAddressPrefix: 'VirtualNetwork'
+          sourcePortRange: '*'
+          destinationAddressPrefix: '*'
+          destinationPortRanges: [
+            '3389'
+            '22'
+          ]
+        }
+      }
+    ]
+    tags: tags
+  }
+}
+
+module virtualNetwork 'br/public:avm/res/network/virtual-network:0.1.8' = if (createVirtualNetwork) {
+  name: '${uniqueString(deployment().name, location)}-virtual-network'
+  params: {
+    name: virtualNetworkSettings.?name ?? 'vnet-${name}'
+    location: location
+    enableTelemetry: enableTelemetry
+    addressPrefixes: [
+      virtualNetworkSettings.?addressPrefix ?? '10.0.0.0/16'
+    ]
+    subnets: union(
+      [
+        {
+          addressPrefix: virtualNetworkSettings.?subnet.addressPrefix ?? '10.0.0.0/24'
+          name: virtualNetworkSettings.?subnet.name ?? 'default'
+          networkSecurityGroupResourceId: createDefaultNsg
+            ? defaultNetworkSecurityGroup.outputs.resourceId
+            : virtualNetworkSettings.?subnet.networkSecurityGroupResourceId
+        }
+      ],
+      createBastion
+        ? [
+            {
+              addressPrefix: bastionSettings.?subnetAddressPrefix ?? '10.0.1.0/26'
+              name: 'AzureBastionSubnet'
+              networkSecurityGroupResourceId: bastionSettings.?networkSecurityGroupResourceId
+            }
+          ]
+        : []
+    )
+    tags: tags
+  }
+}
+
+module bastion 'br/public:avm/res/network/bastion-host:0.2.2' = if (createBastion) {
+  name: '${uniqueString(deployment().name, location)}-bastion-host'
+  params: {
+    name: bastionSettings.?name ?? 'bas-${name}'
+    location: location
+    skuName: bastionSettings.?sku ?? 'Standard'
+    enableTelemetry: enableTelemetry
+    virtualNetworkResourceId: virtualNetwork.outputs.resourceId
+    disableCopyPaste: bastionSettings.?disableCopyPaste
+    enableFileCopy: bastionSettings.?enableFileCopy
+    enableIpConnect: bastionSettings.?enableIpConnect
+    enableKerberos: bastionSettings.?enableKerberos
+    enableShareableLink: bastionSettings.?enableShareableLink
+    scaleUnits: bastionSettings.?scaleUnits
+    tags: tags
+  }
+}
+
+module virtualMachine 'br/public:avm/res/compute/virtual-machine:0.5.3' = if (createVirtualMachine) {
+  name: '${uniqueString(deployment().name, location)}-virtual-machine'
+  params: {
+    name: virtualMachineSettings.?name ?? 'vm-${name}'
+    computerName: take(virtualMachineSettings.?name ?? 'vm-${name}', 15)
```

Comment:
I wonder if it might not be 'safer' to restrict the VM name to only 15 characters in the first place. Sure the 'name' can technically be longer than the 'computerName' - but at the same time just 'taking the first 15 chars' might lead to unexpected results for the user. Thoughts?

Reply:
Good idea, I've added the restriction, however I've moved the `take` to the default computer name. otherwise I would have to restrict the entire `name` param just for this one VM setting.

Reply:
Sounds like a reasonable compromise. Thanks @cecheta. Resolving comment...